docs/gallery/advanced_io/zarr_io.py

```diff
@@ -0,0 +1,89 @@
+"""
+Zarr IO
+=======
+
+Zarr is an alternative backend option for NWB files. It is a Python package that
+provides an implementation of chunked, compressed, N-dimensional arrays. Zarr is a good
+option for large datasets because, like HDF5, it is designed to store data on disk and
+only load the data into memory when needed. Zarr is also a good option for parallel
+computing because it supports concurrent reads and writes.
```

Comment:
I would suggest adding a note here along the lines of: 1) We use the `Zarr.DirectoryStore` or `Zarr.NestedDrirectoryStore` 2) the storage formats are optimized for storage in cloud storage (e.g., S3), 3) for very large files, Zarr will create many files which can lead to issues for traditional file system (that are not cloud object stores) due to limitations on the number of files per directory (this affects local disk, GDrive, Dropbox etc.)